Agent version (#6129)
DataDog/documentation: content/en/agent/basic_agent_usage/_index.md

## Agent Architecture

{{< tabs >}}
{{% tab "Agent v6 & v7" %}}

> Agent v7 and v7 are composed of a main process responsible for collecting infrastructure metrics, logs, and receiving [DogStatsD metrics][1]. The main components to this process are:

Reviewer: ```suggestion
Agent v6 and v7 are composed of a main process responsible for collecting infrastructure metrics, logs, and receiving [DogStatsD metrics][1]. The main components to this process are:
```

| 8125 | Used for the DogStatsD server to receive external metrics.                                  |

### The Collector

> The collector gathers all standard metrics every 15 seconds. Agent v6 embed a Python2.7 interpreter to run integrations and [custom checks][4].

Reviewer 2: There's a typo in the last sentence: "Agent v6 embed". Should be "embeds". Also the statement is not 100% accurate since 6.14+ also embeds a 3.0 interpreter.

> The collector gathers all standard metrics every 15 seconds. Agent v6 embeds a Python2.7 interpreter to run integrations and [custom checks][4].

Reviewer 3: ```suggestion
The collector gathers all standard metrics every 15 seconds. Agent v6 embeds a Python 2.7 interpreter to run integrations and [custom checks][4].
```

### The Forwarder

The Agent forwarder send metrics over HTTPS to Datadog. Buffering prevents network splits from affecting metric reporting. Metrics are buffered in memory until a limit in size or number of outstanding send requests are reached. Afterwards, the oldest metrics are discarded to keep the forwarder's memory footprint manageable. Logs are sent over an SSL-encrypted TCP connection to Datadog.

### DogStatsD

> In v6, DogStatsD is a Golang implementation of [Etsy's StatsD][5] metric aggregation daemon. It is used to receive and roll up arbitrary metrics over UDP or unix socket, thus allowing custom code to be instrumented without adding latency to the mix. Learn more about [DogStatsD][6].

Reviewer 3: ```suggestion
In v6, DogStatsD is a Golang implementation of [Etsy's StatsD][5] metric aggregation daemon. It is used to receive and roll up arbitrary metrics over UDP or Unix socket, thus allowing custom code to be instrumented without adding latency. Learn more about [DogStatsD][6].
```

|----------|-------------------------------------------|
| [AIX][1] | AIX 6.1 TL9 SP6, 7.1 TL5 SP3, 7.2 TL3 SP0 |

[1]: /agent/basic_agent_usage/aix
{{% /tab %}}
{{< /tabs >}}

## CLI

> With Agent v6+ the command line interface is sub-command based. To run a sub-command, first invoke the Agent binary:

Reviewer 3: ```suggestion
With Agent v6+, the command line interface is based on subcommands. To run a subcommand, first invoke the Agent binary:
```

```
<AGENT_BIN_PATH> <SUB_COMMAND> <OPTIONS>
```

> | Sub-Command       | Notes                                                                      |

Reviewer 3: ```suggestion
| Subcommand       | Notes                                                                      |
```

|-------------------|----------------------------------------------------------------------------|
| `check`           | Run the specified check                                                    |
| `configcheck`     | Print all configurations loaded & resolved of a running Agent              |

> | `diagnose`        | Execute some connectivity diagnosis on your system                         |

Reviewer 3: ```suggestion
| `diagnose`        | Execute connectivity diagnosis on your system                         |
```

| `flare`           | [Collect a flare and send it to Datadog][1]                                |
| `health`          | Print the current Agent health                                             |
| `help`            | Help about any command                                                     |
| `hostname`        | Print the hostname used by the Agent                                       |
| `import`          | Import and convert configuration files from previous versions of the Agent |
| `installservice`  | Installs the Agent within the service control manager                      |
| `launch-gui`      | Starts the Datadog Agent GUI                                               |

> | `regimport`       | Import the registry settings into datadog.yaml                             |

Reviewer 3: ```suggestion
| `regimport`       | Import the registry settings into `datadog.yaml`                             |
```

| `remove-service`  | Removes the Agent from the service control manager                         |
| `restart`         | [Restart the Agent][2]                                                     |
| `restart-service` | Restarts the Agent within the service control manager                      |
| `start`           | [Start the Agent][3]                                                       |
| `start-service`   | Starts the Agent within the service control manager                        |
| `status`          | [Print the current Agent status][4]                                        |
| `stop`            | [Stop the Agent][5]                                                        |
| `stopservice`     | Stops the Agent within the service control manager                         |
| `version`         | Print the version info                                                     |

> **Note**: Some options have their own set of flags and options detailed in a help message. For example, to see how to use the `check` sub-command, run:

Reviewer 3: ```suggestion
**Note**: Some options have their own set of flags and options detailed in a help message. For example, to see how to use the `check` subcommand, run:
```
